### DISCUSSION TITLE
Page turning animation

### DISCUSSION THREAD

**Discussion by MobileFreelancer** [readium/mobile#21] [Ideas]
Hello,

I have successfully integrated readium library with iOS swift only thing I want that it will be a great if library should support animations like we are flipping page. Book reading with page turning effect. Is that possible to do some customization regarding this?Anyone have idea related to this then please help me.I tried by my way but not working as it should be.

Thanks.

**Comment by mickael-menu** [comment]
At the moment there's no way to customize page animations in Readium 2 Swift and Kotlin.

For EPUB, this is a complicated feature to implement and I don't see this coming anytime soon.

For PDF and CBZ though, it would be possible to rely on the underlying `UIPageViewController` and `PDFView` APIs.

**Reply by MobileFreelancer** [reply]
I have used EPUB files only and tried different ways to achieve desire animation but it's not happening as I want.Thanks for your reply if it will happen in upcoming versions then it would be great for me.Thanks once again.

**Reply by MobileFreelancer** [reply]
Hello,

I have used R2 navigator framework and customised with spread loaded using PageViewController but animation is applied on whole spread instead of that it should be only on content displayed also not able to achieve mid spine animation in pageViewController when device is in landscape mode.It should split from mid as we actually reading a book.If you can find any solution or tried by using pageViewController then it would really helpful for me.

**Reply by mickael-menu** [reply]
Yes, that's why it's difficult to implement properly for reflowable EPUBs.

**Reply by Seanc949** [reply]
We are ready to invest resources to implement a custom page flip animation.  Can you provide an enhancement spot for it?

**Reply by mickael-menu** [reply]
@Seanc949 It's such a disrupting feature that I doubt it can be implemented as an external contribution. You will need to modify in depth how the Navigator works.

But if you have a plan, feel free to come talk about it on our [`#architecture` Slack channel](http://readium-slack.herokuapp.com/).
